core/system/inc/svc_gw.h

```diff
@@ -55,12 +55,14 @@ static inline uint32_t svc_gw_get_dst_fn(TSecGw *svc_pc)
 
 static inline uint8_t svc_gw_get_dst_id(TSecGw *svc_pc)
 {
-    uint32_t box_id = svc_pc->cfg_ptr - __uvisor_config.cfgtbl_ptr_start;
+    uint8_t box_id;
 
-    if(box_id <= 0 || box_id >= g_vmpu_box_count)
```

Comment:
Thank you for spotting this. Currently we do *not* allow secure gateways to box 0, so the condition must still hold (I'll change the check accordingly for this special case).

@meriac In general, are we sure we don not want to allow secure gateways to box 0? I think this restriction goes against the principle of "box 0 is just a box like all other boxes". Interrupts can be context switched to box 0 normally, for example. If the restriction is not needed for security purposes, I'd rather remove it.